### PR TITLE
Tests for ParameterizedRunSelector and for RunSelectorParameter

### PR DESCRIPTION
**Pull request by alexsomai**
Summary of this pull request:
- [X] Fix and update existing tests for `ParameterizedRunSelector`
- [X] Fix and update existing tests for `RunSelectorParameter`

CC @ikedam @oleg-nenashev @martinda

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/runselector/selectors/ParameterizedRunSelectorTest.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    /**
+     * Should not cause a fatal error even for an undefined variable.
+     */
+    @Issue("JENKINS-30357")
+    @Test
+    public void testUndefinedParameter() throws Exception {
+        FreeStyleProject jobToSelect = j.createFreeStyleProject();
+        FreeStyleProject selecter = j.createFreeStyleProject();
+
+        RunSelector selector = new ParameterizedRunSelector("NoSuchVariable");
```
Comment (ikedam) [inline src/test/java/org/jenkinsci/plugins/runselector/selectors/ParameterizedRunSelectorTest.java]: I think this test is no longer necessary as `ParameterizedRunSelector` now only supports variable expressions and immediate values and no longer supports variable names.

**Comment by ikedam** [comment]
Looks good to me.

File: src/test/java/org/jenkinsci/plugins/runselector/selectors/ParameterizedRunSelectorTest.java (hunk 1 above)
Comment (alexsomai) [inline src/test/java/org/jenkinsci/plugins/runselector/selectors/ParameterizedRunSelectorTest.java]: Yes, you are right

**Comment by oleg-nenashev** [comment]
LGTM 👍

**Pull request merged**